stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt

```diff
@@ -287,11 +288,21 @@ internal class QueryChannelsLogic(
      * @param cidList The channels to refresh.
      */
     private fun refreshChannels(cidList: Collection<String>) {
-        mutableState._channels.value += mutableState.queryChannelsSpec.cids
+        mutableState.queryChannelsSpec.cids
             .intersect(cidList)
-            .associateWith { cid ->
-                val (channelType, channelId) = cid.cidToTypeAndId()
-                stateRegistry.channel(channelType, channelId).toChannel()
+            .map { cid -> cid.cidToTypeAndId() }
+            .filter { (channelType, channelId) ->
+                stateRegistry.isActiveChannel(
+                    channelType = channelType,
+                    channelId = channelId,
+                )
+            }
+            .associate { (channelType, channelId) ->
+                val cid = (channelType to channelId).toCid()
+                cid to stateRegistry.channel(
+                    channelType = channelType,
+                    channelId = channelId,
+                ).toChannel()
             }
```

Comment:
We are not keeping the result anywhere, is it right?

Reply:
good catch! Thank you!